homework-1/task_2.py

```diff
@@ -0,0 +1,72 @@
+
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code.
+Feel free to add any more new features to your class.
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student.
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id, subject):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = {
+            "name": name,
+            "subject":subject
+        }
+
+class CFGstudent(Student):
+
+    def __init__(self, name, age, id, subject):
+        super().__init__()
```

Comment:
Great start here - we needed to call `super().__init__` with the correct arguments (`name`, `age` and `id`), and set the specialisation separately